Mod2/Lessons/Week2/DataRelationships.md

```diff
@@ -0,0 +1,158 @@
+
+<p align='center'>
+  <img src='../../Images/Week2/one_to_many_ERD.png'>
+</p>
+
+Whenever we have a **one to many** relationship, Resource A will always **belong to** Resource B, with the use of a **foreign key**; and, Resource B **has many** Resource A.
+
+### One-to-One
+
+A one-to-one relationship is the least common, it is where a record in one table is associated with exactly one record in another table.
+<p align='center'>
+  <img src='../../Images/Week2/one_to_one_ERD.png'>
+</p>
+
+### Many-to-Many
+In a many-to-many relationship both sides of the relationship reference multiple records.
+
+<p align='center'>
+  <img src='../../Images/Week2/many_to_many_ERD.png'>
+</p>
+
+## Implementing a Many-To-Many Relationship
+
+<p align='center'>
+  <img src='../../Images/Week2/many_to_many_bad.png'>
+</p>
+
+So far, we have used foreign keys to create relationships. The problem is that a foreign key can identify a single record from another table, but in a many-to-many both sides of the relationship need to reference multiple records. This means that we’re going to need more than just foreign keys.
+
+One solution that might come to mind is, instead of our foreign keys storing a single id, we could store an array of ids. This is a good guess, but in a database there is no concept of an array. The reason is that, for the database to be efficient, it needs to know the exact size of the datatype it is storing. For example, our DB knows exactly how much space an integer takes up (8 bytes).
+
+But what about strings? They can vary in length. The Database handles strings by giving them a maximum length, which is 255 bytes by default. If we tried to do the same thing with arrays, we would be limiting how many relationships we could create, which is a bigger problem than limiting how long a string can be.
+
+The key takeaway here is a database can’t store an array of ids, so we’re going to need something else.
+
+
+## Join Tables
+Since we can’t achieve the many-to-many relationship with our given tables, we are going to add a third table to manage this relationship. This is called a join table.
+
+<p align='center'>
+  <img src='../../Images/Week2/many_to_many_good.png'>
+</p>
+
+Note: Join tables are just ordinary tables with a unique purpose. Each row of our join table relates a row of one table to a row of another table.
+
+Turn and Talk: Which songs are associated with which playlists?
+
+The join table’s name doesn’t really matter. We could call it Song_Playlist or Playlist_Song, it’s really up to you as the developer. You could even choose to name it happy_fun_times but that would be confusing.
+
+When you’re thinking about what to call this table, think about how you’re likely to use it most within your application. Since our app’s goal will be to show a playlist of songs more often, we’re going to call it playlist_songs.
```

Comment:
Is there a convention we should follow?

Reply:
I did a bit of googling and it seems like everyone is doing something different. I might put some of the other examples in here to show students there isn't as set as some other things we teach. 

This is what the entity relationship diagram folks I took the later example from use. I actually should update this last place you pointed out. I changed this from what backend was doing with calling the join table PlaylistSongs. I think underscores is more common for table names? If you know of a better source for SQL naming conventions let me know 🤷